### PR TITLE
[SPARK-46390] Update the `Example` page

### PR DESCRIPTION
**Pull request by zhengruifeng**
1, rewrite the outdated `RDD API examples` with dataframe APIs;
2, update `DataFrame API examples` and `Machine learning example`:

- `sc.textFile(...).map(...).toDF` -> `spark.read.text`
- `sqlContext.read` -> `spark.read`
- `sqlContext.createDataFrame` -> `spark.createDataFrame`

image

image

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
I want to make a few more modification, let me convert it to draft first

**Review by HyukjinKwon — APPROVED** [review]

File: examples.md [hunk 1]
```diff
@@ -19,12 +19,18 @@ On top of Spark’s RDD API, high level APIs are provided, e.g.
...
+In Spark, a <a href="https://spark.apache.org/docs/latest/sql-programming-guide.html#dataframes">DataFrame</a>
+is a distributed collection of data organized into named columns.
+Users can use DataFrame API to perform various relational operations on both external
+data sources and Spark’s built-in distributed collections without providing specific procedures for processing data.
+Also, programs based on DataFrame API will be automatically optimized by Spark’s built-in optimizer, Catalyst.
+</p>
 
 <h3>Word count</h3>
-<p>In this example, we use a few transformations to build a dataset of (String, Int) pairs called <code>counts</code> and then save it to a file.</p>
+<p>In this example, we use a few transformations to build a dataset of (String, Long) pairs and then save it to a file.</p>
```
Comment (viirya) [inline examples.md:33]: Hmm, I think `transformations` are terms used for RDD operations (i.e, `transformations` and `actions` as this page mentioned above)? For DataFrame, maybe `relational operations`?

File: examples.md [hunk 2]
```diff
@@ -36,36 +42,38 @@ In this page, we will show examples using RDD API as well as examples using high
...
 <div class="tab-pane tab-pane-java">
 <div class="code code-tab">
 {% highlight java %}
-JavaRDD<String> textFile = sc.textFile("hdfs://...");
-JavaPairRDD<String, Integer> counts = textFile
-    .flatMap(s -> Arrays.asList(s.split(" ")).iterator())
-    .mapToPair(word -> new Tuple2<>(word, 1))
-    .reduceByKey((a, b) -> a + b);
-counts.saveAsTextFile("hdfs://...");
+DataFrame df = spark.read.text("hdfs://...").toDF("text");
```
Comment (viirya) [inline examples.md:71]: Hmm, for Java, isn't it `read()`?
```suggestion
DataFrame df = spark.read().text("hdfs://...").toDF("text");
```

File: examples.md [hunk 3]
```diff
@@ -186,13 +178,7 @@ errors.filter(col("line").like("%MySQL%")).collect()
...
 {% highlight java %}
 // Creates a DataFrame having a single column named "line"
-JavaRDD<String> textFile = sc.textFile("hdfs://...");
-JavaRDD<Row> rowRDD = textFile.map(RowFactory::create);
-List<StructField> fields = Arrays.asList(
-  DataTypes.createStructField("line", DataTypes.StringType, true));
-StructType schema = DataTypes.createStructType(fields);
-DataFrame df = sqlContext.createDataFrame(rowRDD, schema);
-
+DataFrame df = spark.read.text("hdfs://...").toDF("line");
```
Comment (viirya) [inline examples.md:181]:
```suggestion
DataFrame df = spark.read().text("hdfs://...").toDF("line");
```

File: examples.md [hunk 4]
```diff
@@ -36,36 +42,38 @@ In this page, we will show examples using RDD API as well as examples using high
...
-    .flatMap(s -> Arrays.asList(s.split(" ")).iterator())
-    .mapToPair(word -> new Tuple2<>(word, 1))
-    .reduceByKey((a, b) -> a + b);
-counts.saveAsTextFile("hdfs://...");
+DataFrame df = spark.read.text("hdfs://...").toDF("text");
+
+df.select(explode(split(col("text"), " ")).alias("word"))
+  .groupBy("word")
+  .agg(count(lit(0)).alias("count"))
+  .write.parquet("hdfs://...");
```
Comment (viirya) [inline examples.md:76]:
```suggestion
df.select(explode(split(col("text"), " ")).alias("word"))
  .groupBy("word")
  .agg(count(lit(0)).alias("count"))
  .write().parquet("hdfs://...");
```

**Comment by zhengruifeng** [comment]
@viirya @srowen @HyukjinKwon thank you so much for the reviews.

FYI, @MrPowers is preparing a new version of this page with much better examples, I am going to close this PR in favor of the new one

**Pull request closed without merging**